### DISCUSSION TITLE
How to build project targeting Net35 with GitHub Actions

### DISCUSSION THREAD

**Discussion by krafs** [community/community#25825] [Actions]
<p>I have a project that targets .NET Framework 3.5. How do I build it in a GitHub Actions workflow?<br>
There is an action for setting up a <a href="https://github.com/actions/setup-dotnet" rel="nofollow noopener">dotnet core</a> environment: Is there something similar to Net35? Or some other way to do it?</p>

**Comment by eddiechenms** [comment, marked as answer]
<p>You can call " <strong>msbuild.exe</strong>" which is installed on “Windows Server 2019” and “Windows Server 2016 R2” virutal environments to build .Net Framework projects.</p>
<p>Here is the example to call msbuild to build the project with “Windows Server 2019” runner:</p>
<pre><code>jobs:
  build:
    runs-on: windows-2019
    steps:
      - name: checkout
        uses: actions/checkout@v1
      - name: Build DotNET35
        run: |
           cd "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\"
           .\MSBuild.exe $Env:GITHUB_WORKSPACE\pathtoyoursolutionorproject
</code></pre>
<p>If you want to run the workflow with “Windows Server 2016 R2”, change the path to:</p>
<pre><code>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\
</code></pre>

**Reply by ghost** [reply]
According to 
`https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md`
.NET 3.5 SDK is also installed - along with other versions - in ubuntu-20.04, but I couldn't configure an action to use .NET 3.5 by default, so whenever it runs, it picks dotnet 7.0.102. 
Does anyone know how select it?
